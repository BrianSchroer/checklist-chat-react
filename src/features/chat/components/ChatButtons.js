import React, {PropTypes} from 'react';

const ChatButtons = ({onChatMessageAddRequest, onChatParticipantsRequest}) => {
    return(
        <div>
            <button className="btn btn-primary"
                onClick={onChatMessageAddRequest}>Say something...</button>
            <button className="btn btn-default pull-right"
                onClick={onChatParticipantsRequest}>Who's here?</button>
        </div>
    );
};

ChatButtons.propTypes = {
    onChatMessageAddRequest: PropTypes.func.isRequired,
    onChatParticipantsRequest: PropTypes.func.isRequired
};

export default ChatButtons;
