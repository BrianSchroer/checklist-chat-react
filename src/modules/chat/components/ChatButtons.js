import React from 'react';
import PropTypes from 'prop-types';

const ChatButtons = ({onChatMessageAddRequest, onChatParticipantsRequest}) => {
    return(
        <div>
            <button className="btn btn-primary"
                onClick={onChatMessageAddRequest}>Say something...</button>
            <button className="btn btn-default pull-right"
                onClick={onChatParticipantsRequest}>Who's here?</button>
        </div>
    );
};

ChatButtons.propTypes = {
    onChatMessageAddRequest: PropTypes.func.isRequired,
    onChatParticipantsRequest: PropTypes.func.isRequired
};

export default ChatButtons;
