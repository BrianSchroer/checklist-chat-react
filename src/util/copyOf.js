export default original => Object.assign({}, original);
