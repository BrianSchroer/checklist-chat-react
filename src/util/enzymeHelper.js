const enzymeHelper = class {

    static nestedFind(wrapper, selector) {
        const selectors = selector.split('>').map(node => node.trim());
        let parent = wrapper;
        let found = [];

        for (let i = 0; i < selectors.length; i++) {
            found = parent.find(selectors[i]);
            if (!found.length) {
                break;
            }
        }

        return found;
    }

    static findSingle(wrapper, selector) {
        const findArray = enzymeHelper.nestedFind(wrapper, selector);

        if (findArray.length != 1) {
            throw(`expected 1 find for selector "${selector}", but found ${findArray.length}.`);
        }

        return findArray.first();
    }
};

export default enzymeHelper;
