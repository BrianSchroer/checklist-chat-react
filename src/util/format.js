import moment from 'moment';

export default class {
    static time(input = '') {

        const isoDateFormatRegex = /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}([+-]\d{2}:\d{2})?/g;
        let formatted = input;

        if (input && input.match(isoDateFormatRegex)) {
            formatted = this.dateTimeString(input, 'h:mm a');
        }

        return formatted;
    }

    static dateTimeString(input, formatString) {
        let formatted = input;

        if (input) {
            const testMoment = moment(input);

            if (testMoment.isValid()) {
                const testFormat = testMoment.format(formatString);
                if (testFormat != 'Invalid date') {
                    formatted = testFormat;
                }
            }
        }

        return formatted;
    }
}
