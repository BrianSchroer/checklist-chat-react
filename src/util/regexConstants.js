export const CSS_PATH = /(\.css)$/;
export const EOT_PATH = /\.eot(\?v=\d+\.\d+\.\d+)?$/;
export const JS_PATH = /\.js$/;
export const SVG_PATH = /\.svg(\?v=\d+\.\d+\.\d+)?$/;
export const TTF_PATH = /\.ttf(\?v=\d+\.\d+\.\d+)?$/;
export const WOFF_OR_WOFF2_PATH = /\.(woff|woff2)(\?v=\d+\.\d+\.\d+)?$/;
