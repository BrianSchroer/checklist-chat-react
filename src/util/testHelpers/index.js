import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import enzymeHelper from './enzymeHelper';
import testRendererHelper from './testRendererHelper';
import testStoreHelper from './testStoreHelper';

export {
    renderer as renderer,
    shallow as shallow,
    mount as mount,
    enzymeHelper as enzymeHelper,
    testRendererHelper as testRendererHelper,
    testStoreHelper as testStoreHelper
};
