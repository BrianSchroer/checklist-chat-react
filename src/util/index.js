import format from './format';
import uiHelper from './uiHelper';
import validationHelper from './validationHelper';

export {
    format as format,
    uiHelper as uiHelper,
    validationHelper as validationHelper
};
