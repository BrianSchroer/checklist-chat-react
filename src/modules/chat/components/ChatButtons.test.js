import React from 'react';
import {testRendererHelper, shallow, enzymeHelper} from '../../../util/testHelpers';
import ChatButtons from './ChatButtons';

const defaultProps = {
    onChatMessageAddRequest: () => {},
    onChatParticipantsRequest: () => {}
};

function shallowRender(propOverrides) {
    const props = Object.assign({}, defaultProps, propOverrides);
    return shallow(<ChatButtons {...props} />);
}

describe('ChatButtons', () => {
    it('should render correctly', () => {
        testRendererHelper.expectSnapshotMatch(<ChatButtons {...defaultProps}/>);
    });

    it('should handle chat message add request', () => {
        let wasCalled = false;

        const button = enzymeHelper.findSingle(
            shallowRender({ onChatMessageAddRequest: () => {wasCalled = true;} }),
            'div > button.btn-primary');

        button.simulate('click');

        expect(wasCalled).toBe(true);
    });

    it('should handle chat participants request', () => {
        let wasCalled = false;

        const button = enzymeHelper.findSingle(
            shallowRender({ onChatParticipantsRequest: () => {wasCalled = true;} }),
            'div > button.btn-default');

        button.simulate('click');

        expect(wasCalled).toBe(true);
    });
});
