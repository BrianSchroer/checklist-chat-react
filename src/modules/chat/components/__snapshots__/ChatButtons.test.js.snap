// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatButtons should render correctly 1`] = `
<div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Say something...
  </button>
  <button
    className="btn btn-default pull-right"
    onClick={[Function]}
  >
    Who's here?
  </button>
</div>
`;
